design/reconciler-patterns.md

```diff
@@ -0,0 +1,653 @@
+# Crossplane Reconciler Patterns
+
+While Kubernetes declarative style API probably one of the best features of Kubernetes as a platform, Kubernetes API extensibility is arguably one of the most exciting ones. 
+Using Kubernetes Operators patters user can extend Kubernetes API types to support Kubernetes internal or external resources. 
+Controller-Runtime (defacto standard for Kubernetes operator frameworks: Kubebuilder, Operator SDK) does a great job simplifying the development process for Kubernetes operator by providing default project scaffolding and infrastructure libraries plumbing. Nevertheless, using a Declarative Style API (Kubernetes) to perform Imperative API operation (Managed Cloud Resources) is less than a straight-forward process.  
+
+
+## Goals
+
+Using crossplane experience through trial and errors summarize existing paradigms and come up with the unified paradigm recommendation for writing crossplane controllers. 
+
+
+## Non-goals
+
+This document is intended to be at best a guideline for best practices for writing crossplane operators at this point in time. 
+It is by no means a prescriptive standard to which all others must follow, rather it is merely a recommendation. 
+
+
+## Terminology
+
+*   _Crossplane Object_ refers to the CRD Type manifestation in Kubernetes API.
+    *   This term is used interchangeably with Kubernetes object or object
```

Comment:
In general, I agree, however, since I don't think I refer to them as such in this document, do you still think it worth including this disambiguation?